### PR TITLE
[TS-1633] Added support of th2 transport protocol

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -60,28 +51,22 @@ java {
...
-    implementation group: 'org.jetbrains.kotlin', name: 'kotlin-reflect'
-    implementation group: 'io.github.microutils', name: 'kotlin-logging', version: '1.7.9'
-    implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-kotlin', version: '2.11.2'
-    implementation group: 'com.athaydes.rawhttp', name:'rawhttp-core', version: '2.4.1'
+    implementation group: 'io.github.microutils', name: 'kotlin-logging', version: '3.0.5'
+    implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-kotlin', version: '2.15.2'
+    implementation group: 'com.athaydes.rawhttp', name:'rawhttp-core', version: '2.6.0'
 
-    testImplementation group: 'org.jetbrains.kotlin', name: 'kotlin-test-junit5'
+    testImplementation group: 'org.jetbrains.kotlin', name: 'kotlin-test-junit5', version: kotlin_version
```
Comment (OptimumCode) [inline build.gradle:69]: We don't have to specify version for kotlin dependencies because kotiln plugin does it by itself

File: build.gradle [hunk 2]
```diff
@@ -60,28 +51,22 @@ java {
...
-    implementation 'org.slf4j:slf4j-api:1.7.32'
+    implementation 'org.slf4j:slf4j-api'
 
-    implementation group: 'org.jetbrains.kotlin', name: 'kotlin-stdlib-jdk8'
-    implementation group: 'org.jetbrains.kotlin', name: 'kotlin-reflect'
-    implementation group: 'io.github.microutils', name: 'kotlin-logging', version: '1.7.9'
-    implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-kotlin', version: '2.11.2'
-    implementation group: 'com.athaydes.rawhttp', name:'rawhttp-core', version: '2.4.1'
+    implementation group: 'io.github.microutils', name: 'kotlin-logging', version: '3.0.5'
+    implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-kotlin', version: '2.15.2'
```
Comment (OptimumCode) [inline build.gradle:66]: Do we nee a version here? This should be taken from the BOM I think

File: build.gradle
Comment (OptimumCode) [inline build.gradle:1]: Shouldn't we add licenses check, vulnerabilities check and git properties plugins?

File: src/main/kotlin/com/exactpro/th2/httpws/client/Main.kt [hunk 3]
```diff
@@ -256,6 +408,7 @@ data class WebSocketSettings(
         TEXT {
             override fun send(client: IClient, data: ByteArray) = client.sendText(data.toString(UTF_8))
         },
+        @Suppress("unused")
```
Comment (OptimumCode) [inline src/main/kotlin/com/exactpro/th2/httpws/client/Main.kt:411]: Please, remove it... If we don't use it right now it does not mean we should suppress the warning. I would better have a warning then forget about this enum value when I looking for unused code (if I am doing refactoring or something)

**Review by OptimumCode — COMMENTED** [review]
We probably should rename workflow files as they do not publish anything to the Sonatype

File: build.gradle [hunk 4]
```diff
@@ -1,6 +1,14 @@
...
 plugins {
     id 'com.palantir.docker' version '0.25.0' apply false
     id 'org.jetbrains.kotlin.jvm' version "${kotlin_version}"
+    id "org.owasp.dependencycheck" version "8.3.1"
+    id "com.gorylenko.gradle-git-properties" version "2.4.1"
+    id 'com.github.jk1.dependency-license-report' version '2.5'
+    id "de.undercouch.download" version "5.4.0"
 }
 
 apply plugin: 'kotlin'
```
Comment (OptimumCode) [inline build.gradle:14]: We can apply all the plugins in `plugins` block

File: build.gradle [hunk 5]
```diff
@@ -102,16 +95,37 @@ docker {
     copySpec.from(tarTree("$buildDir/distributions/${applicationName}.tar"))
 }
 
-compileKotlin {
-    kotlinOptions {
-        jvmTarget = "11"
-        freeCompilerArgs += "-XXLanguage:+UnitConversion"
+tasks.withType(KotlinCompile).configureEach {
+    kotlinOptions.jvmTarget = "11"
```
Comment (OptimumCode) [inline build.gradle:99]: This block is deprecated in the new kotlin version. The `compilerOptions` should be used instead
```groovy
tasks.withType(KotlinCompile).configureEach {
    compilerOptions {
        jvmTarget.set(JvmTarget.JVM_11)
    }
}
```

**Review by OptimumCode — DISMISSED** [review]

**Review by OptimumCode — APPROVED** [review]

**Comment by Nikita-Smirnov-Exactpro** [comment]
I'm going to add split session setting feature

**Pull request merged**